### PR TITLE
Refs 713 : Genre Detail 

### PR DESCRIPTION
**Pull request by HadesPTIT**
https://edu-redmine.framgia.vn/issues/713
Fix fetch data + Genre detail screen
detail screen

### CODE REVIEW

**Comment by HadesPTIT** [comment]
ready

**Comment by HadesPTIT** [comment]
ready

**Comment by HadesPTIT** [comment]
ready

File: app/src/main/java/com/framgia/soundcloudproject/screen/main/home/GenreAdapter.java [hunk 1]
```diff
@@ -63,4 +73,18 @@ public void bindView(Genre genre) {
...
+    }
+
+    public void setOnGenreClickListener(OnGenreClickListener listener) {
+        mListener = listener;
+    }
+
+    interface OnGenreClickListener {
+        void onGenreClicked(String genre);
+    }
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/soundcloudproject/screen/main/home/GenreAdapter.java]: remove bl

File: app/src/main/java/com/framgia/soundcloudproject/screen/main/home/GenreAdapter.java [hunk 2]
```diff
@@ -46,14 +50,20 @@ public int getItemCount() {
         return mGenres != null ? mGenres.size() : 0;
     }
 
-    static class ViewHolder extends RecyclerView.ViewHolder {
+    public class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloudproject/screen/main/home/GenreAdapter.java]: keep holder is static

File: app/src/main/java/com/framgia/soundcloudproject/screen/main/home/HomeFragment.java [hunk 3]
```diff
@@ -35,21 +42,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         mRecyclerView = view.findViewById(R.id.recyclerview_genre);
         mRecyclerView.setLayoutManager(new LinearLayoutManager(mContext));
 
+        mGenreAdapter = new GenreAdapter(mContext, null);
+        mRecyclerView.setAdapter(mGenreAdapter);
+        mGenreAdapter.setOnGenreClickListener(new GenreAdapter.OnGenreClickListener() {
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/soundcloudproject/screen/main/home/HomeFragment.java]: nên implement interface cho đẹp code

File: app/src/main/java/com/framgia/soundcloudproject/screen/main/home/HomeFragment.java [hunk 4]
```diff
@@ -35,21 +42,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-        GenreAdapter adapter = new GenreAdapter(mContext, genres);
-        mRecyclerView.setAdapter(adapter);
+        mGenreAdapter.setGenres(genres);
     }
 
-    @Override
-    public void showSongsByGenre() {
-        // TODO Show list song by genre
+    private void gotoDetailFragment(String genre) {
+        GenreDetailFragment detailFragment = GenreDetailFragment.newInstance(genre);
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/soundcloudproject/screen/main/home/HomeFragment.java]: nên dùng lazy load

Comment (HadesPTIT) [reply app/src/main/java/com/framgia/soundcloudproject/screen/main/home/HomeFragment.java]: E chưa hiểu lắm đoạn này ạ @tuannguyen38

**Comment by HadesPTIT** [comment]
ready

**Pull request merged**